scanner/matcher/updater/vuln/updater.go

```diff
@@ -185,6 +195,77 @@ func validate(opts Opts) error {
 	return nil
 }
 
+func (u *Updater) Import(ctx context.Context, in io.Reader) (err error) {
+	ctx = zlog.ContextWithValues(ctx, "component", "matcher/updater/Updater.Import")
+	iter, iterErr := u.iterateFunc(in)
+	// For each update operation in the JSON blob file.
+	iter(func(op *driver.UpdateOperation, it jsonblob.RecordIter) bool {
+		var ops map[string][]driver.UpdateOperation
+		ops, err = u.store.GetUpdateOperations(ctx, op.Kind, op.Updater)
+		if err != nil {
+			return false
+		}
+		for _, o := range ops[op.Updater] {
+			// This only helps if updaters don't keep something that
+			// changes in the fingerprint.
+			if o.Fingerprint == op.Fingerprint {
+				zlog.Info(ctx).
+					Str("updater", op.Updater).
+					Msg("fingerprint match, skipping")
+				return true
+			}
+		}
+		var ref uuid.UUID
+		count := 0
+		switch op.Kind {
+		case driver.VulnerabilityKind:
+			ref, err = u.store.UpdateVulnerabilitiesIter(ctx, op.Updater, op.Fingerprint, func(yield func(*claircore.Vulnerability, error) bool) {
+				// For each vulnerability in the update operation.
+				it(func(v *claircore.Vulnerability, _ *driver.EnrichmentRecord) bool {
+					count++
+					// Offer one vulnerability to the datastore iterator.
+					return yield(v, nil)
+				})
+				if err := iterErr(); err != nil {
+					yield(nil, err)
+				}
+			})
+		case driver.EnrichmentKind:
+			ref, err = u.store.UpdateEnrichmentsIter(ctx, op.Updater, op.Fingerprint, func(yield func(*driver.EnrichmentRecord, error) bool) {
+				// For each enrichment in the update operation.
+				it(func(_ *claircore.Vulnerability, e *driver.EnrichmentRecord) bool {
+					count++
+					// Offer one enrichment to the datastore iterator.
+					return yield(e, nil)
+				})
+				if err := iterErr(); err != nil {
+					yield(nil, err)
+				}
+			})
+		default:
+			zlog.Warn(ctx).Str("kind", string(op.Kind)).Msg("unknown kind, skipping")
+		}
+		if err != nil {
+			err = fmt.Errorf("updating %s: %w", op.Kind, err)
+			return false
+		}
+		zlog.Info(ctx).
+			Str("updater", op.Updater).
+			Str("kind", string(op.Kind)).
+			Str("ref", ref.String()).
+			Int("count", count).
+			Msg("update imported")
+		return true
+	})
+	if err := iterErr(); err != nil {
+		return fmt.Errorf("iterating on the reader: %w", err)
+	}
+	if err != nil {
+		return err
```

Comment:
nit: maybe add context here, too. iterating over some update operation

Reply:
Any other failure that is not part of the iteration would be captured here. I don't know how to categorize that other than pointing out the fact that this is not due to iterator failures, which would simply negate the previous context (e.g., "not iteration on the reader"). I don't feel like this would too helpful.